draft-ietf-ppm-dap.md

```diff
@@ -1524,82 +1568,108 @@ the requirements in {{request-authentication}}.
 
-To initiate collection, the collector issues a POST request to
-`[leader]/collect`, where `[leader]` is the leader's endpoint URL. The body of
-the request is structured as follows:
+This resource allows the Collector to create new collect jobs in the Leader and
+check on their state. Only the Leader supports this resource.
 
-[OPEN ISSUE: Decide if and how the collector's request is authenticated. If not,
-then we need to ensure that collect job URIs are resistant to enumeration
-attacks.]
+#### Required Methods
+
+##### PUT `/tasks/{task-id}/collect_jobs` {#collect-job-put}
+
+To initiate collection, the Collector issues a PUT request to the collect jobs
+resource. The request body is a `CollectReq`:
 
 ~~~
 struct {
-  TaskID task_id;
   Query query;
   opaque agg_param<0..2^32-1>; /* VDAF aggregation parameter */
 } CollectReq;
 ~~~
 
-The named parameters are:
+[OPEN ISSUE: Decide if and how the collector's request is authenticated. If not,
+then we need to ensure that collect job URIs are resistant to enumeration
+attacks.]
 
-* `task_id`, the DAP task ID. If the Leader does not recognize the task ID, it
-  MUST abort with error `unrecognizedTask`.
 * `query`, the Collector's query. The indicated query type MUST match the task's
-  query type. Otherwise, the Leader MUST abort with error "queryMismatch".
+  query type. Otherwise, the Leader MUST abort with error `queryMismatch`.
 * `agg_param`, an aggregation parameter for the VDAF being executed. This is the
-  same value as in `AggregateInitializeReq` (see {{leader-init}}).
+  same value as in `AggregationJob` (see {{aggregation-job-representation}}).
 
-Depending on the VDAF scheme and how the leader is configured, the leader and
-helper may already have prepared a sufficient number of reports satisfying the
+Depending on the VDAF scheme and how the Leader is configured, the Leader and
+Helper may already have prepared a sufficient number of reports satisfying the
 query and be ready to return the aggregate shares right away, but this cannot be
 guaranteed. In fact, for some VDAFs, it is not be possible to begin preparing
 inputs until the collector provides the aggregation parameter in the
 `CollectReq`. For these reasons, collect requests are handled asynchronously.
 
-Upon receipt of a `CollectReq`, the leader begins by checking that the request
-meets the requirements of the batch parameters using the procedure in
-{{batch-validation}}. If so, it immediately sends the collector a response with
-HTTP status 303 See Other and a Location header containing a URI identifying the
-collect job that can be polled by the collector, called the "collect job URI".
+Upon receipt of a `CollectReq`, the Leader begins by checking that the query is
+valid using the procedure in {{batch-validation}}. If so, it immediately sends
+the Collector a response with HTTP status 303 See Other and a Location header
+containing a URI for the individual collect job that can be polled by the
```

Comment:
Since we are defining URIs for everything else, is there any reason not to introduce a `CollectJobID` and treat collect jobs as we do e.g. reports or aggregation jobs? (i.e. define a specific URI scheme, including the collect job ID, something like `/tasks/{task-id}/collect_jobs/{collect-job-id}`)

This would match the rest of the API much more closely, and I don't think there are any downsides. The collector would be generating the collect job IDs, but I don't think that is a problem.

Reply:
Yeah the argument from consistency seems reasonable. The counter-argument is that not specifying the collect job URI is the least change from DAP-03, and thus the least disruptive to implementations. I can go either way on this. I'm curious what @cjpatton thinks.

Reply:
I hear you; my thoughts are that we should aim to design things as they "should be" if we were writing the design down for the first time, without necessarily worrying about the size of the diff against previous drafts.

Reply:
OK, that sounds like a consensus.

- [ ] specify a `CollectionJobId` and collection job URIs